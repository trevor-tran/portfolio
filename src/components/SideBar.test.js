import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SideBar from './SideBar';

describe("SideBar component", () => {

  it("verifies against its snapshot", () => {
    const mockSectionId = "sectionId";

    document.body.innerHTML =
    `<div id=${mockSectionId}_link></div>`;

    const tree = renderer
      .create(<SideBar sectionId={mockSectionId}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})