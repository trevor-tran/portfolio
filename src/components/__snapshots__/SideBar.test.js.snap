// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component verifies against its snapshot 1`] = `
<div
  className="navbar-container"
>
  <p
    className="h1"
  >
    Trevor Tran
  </p>
  <p
    className="h4"
  >
    Software Engineer
  </p>
  <p>
    Motivated to learn and grow, ready to make an impact as a Software Engineer
  </p>
  <nav
    className="navbar"
  >
    <ul>
      <li>
        <a
          className="navbar-link"
          id="about_link"
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="navbar-link-indicator"
          />
          <span>
            about
          </span>
        </a>
      </li>
      <li>
        <a
          className="navbar-link"
          id="experience_link"
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="navbar-link-indicator"
          />
          <span>
            experience
          </span>
        </a>
      </li>
      <li>
        <a
          className="navbar-link"
          id="projects_link"
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="navbar-link-indicator"
          />
          <span>
            projects
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
